monkey/infection_monkey/master/ip_scanner.py

```diff
@@ -0,0 +1,69 @@
+import logging
+import queue
+import threading
+from queue import Queue
+from threading import Event
+from typing import Callable, Dict, List
+
+from infection_monkey.i_puppet import IPuppet, PingScanData, PortScanData
+
+from .threading_utils import create_daemon_thread
+
+logger = logging.getLogger()
+
+Callback = Callable[[str, PingScanData, Dict[int, PortScanData]], None]
+
+
+class IPScanner:
+    def __init__(self, puppet: IPuppet, num_workers: int):
+        self._puppet = puppet
+        self._num_workers = num_workers
+
+    def scan(self, ips_to_scan: List[str], options: Dict, results_callback: Callback, stop: Event):
```

Comment:
This method has 4 params. options, results_callback and stop seem to be used in almost every method here, maybe they should be object parameters?

Reply:
Say I'm creating a new master. How would I know what is the structure of the `options` and what I need to pass?

Reply:
> This method has 4 params. options, results_callback and stop seem to be used in almost every method here, maybe they should be object parameters?

What would you name it?

> Say I'm creating a new master. How would I know what is the structure of the `options` and what I need to pass?

There's some broader work to be done where we take the config from the Island and transform it into an object instead of passing dicts all over. I was planning on waiting until the dust settled on some of the refactoring in order to avoid unnecessary rework.